Assignment.txt

```diff
@@ -0,0 +1,15 @@
+Armstrong Park This is my theory homework
+1.
+a.O(n)
+b.2(n)
+c.n^2
```

Comment:
Not quite. The first for loop gives a time complexity of n since you iterate through n elements. The second for loop would normally also be n, but since it multiplies the j by two each time, that means you're eliminating half the number of elements each time - this causes a time complexity of log (n). And since they're nested, you multiply the two with each other so that the time complexity is O(nlog(n)).